lib/Dialect/RNS/IR/RNSTypes.cpp

```diff
@@ -60,11 +82,42 @@ struct PolynomialRNSBasisTypeInterface
   }
 };
 
+struct ModArithRNSBasisTypeInterface
+    : public RNSBasisTypeInterface::ExternalModel<ModArithRNSBasisTypeInterface,
+                                                  ModArithType> {
+  bool isCompatibleWith(Type type, Type otherRnsBasisType) const {
+    auto thisType = mlir::dyn_cast<ModArithType>(type);
+    if (!thisType) {
+      return false;
+    }
+
+    auto other = mlir::dyn_cast<ModArithType>(otherRnsBasisType);
+    if (!other) {
+      return false;
+    }
+
+    APInt thisModulus = thisType.getModulus().getValue();
+    APInt otherModulus = other.getModulus().getValue();
+
+    // require same bitwidth
+    if (thisModulus.getBitWidth() != otherModulus.getBitWidth()) {
```

Comment:
I'm slightly surprised this suffices to catch when `getModulus().getType()` are different, since I thought the APInt uses a minimum of 64 bits for its storage type, no matter the underlying value. I see the test that checks this, but I would also wonder if checking `getModulus().getType()` equality here would be safest.